zuul-core/src/main/java/com/netflix/zuul/filters/BaseFilter.java

```diff
@@ -126,15 +126,15 @@ public HttpContent processContentChunk(ZuulMessage zuulMessage, HttpContent chun
     @Override
     public void incrementConcurrency() throws ZuulFilterConcurrencyExceededException {
         final int limit = filterConcurrencyLimit.get();
-        if ((concurrencyProtectEnabled.get()) && (concurrentCount.get() >= limit)) {
+        if ((concurrencyProtectEnabled.get()) && (concurrentCount.longValue() >= limit)) {
```

Comment:
I think `LongAdder` is a poor fit here since the sum needs to be calculated before every increment